### PR TITLE
URGENT - Several Fixes

### PR DESCRIPTION
**Pull request by mathieucarbou**
### Required for https://github.com/Terracotta-OSS/terracotta-platform/pull/587

### CODE REVIEW

File: common/src/main/java/org/terracotta/angela/common/TerracottaCommandLineEnvironment.java [hunk 1]
```diff
@@ -66,10 +66,12 @@ private void validate(String javaVersion, Set<String> javaVendors, Set<String> j
     requireNonNull(javaVendors);
     requireNonNull(javaOpts);
 
-    javaVendors.stream()
-        .filter(((Predicate<String>) vendor -> vendor == null || vendor.isEmpty()).negate())
-        .findFirst()
-        .orElseThrow(() -> new IllegalArgumentException("None of the java vendors can be null or empty"));
+    if(!javaVendors.isEmpty()) {
```
Comment (saurabhagas) [inline common/src/main/java/org/terracotta/angela/common/TerracottaCommandLineEnvironment.java]: This change should not be needed if the `angela.java.vendor` property is correctly set to empty string (not null, empty).

File: common/src/main/java/org/terracotta/angela/common/topology/Version.java [hunk 2]
```diff
@@ -35,7 +35,24 @@ public static Version version(String version) {
     return new Version(version);
   }
 
+  public static Version rawVersion(String version) {
```
Comment (saurabhagas) [inline common/src/main/java/org/terracotta/angela/common/topology/Version.java]: The distribution controller angela creates are based on these versions.

We had explicitly added support for `terracotta-5.7.zip` kit and agreed that we'd use this name for the new kit. I see this changes as both unnecessary and dangerous.

File: common/src/main/java/org/terracotta/angela/common/TerracottaCommandLineEnvironment.java (hunk 1 above)
Comment (mathieucarbou) [reply common/src/main/java/org/terracotta/angela/common/TerracottaCommandLineEnvironment.java]: this is what I did.  I debugged it. There is an empty string, so the validation is failing.

Comment (mathieucarbou) [reply common/src/main/java/org/terracotta/angela/common/TerracottaCommandLineEnvironment.java]: Seems like you didn't try ;-)
`TerracottaCommandLineEnvironment` at the top:

`Set<String> vendors = JAVA_VENDOR.getValue().equals("") ? new HashSet<>() : singleton(JAVA_VENDOR.getValue());`

So if javaVendors, no value is found and you end up in the orElseThrow()

Comment (mathieucarbou) [reply common/src/main/java/org/terracotta/angela/common/TerracottaCommandLineEnvironment.java]: Note: the bug is also there for the java version

File: common/src/main/java/org/terracotta/angela/common/topology/Version.java (hunk 2 above)
Comment (mathieucarbou) [reply common/src/main/java/org/terracotta/angela/common/topology/Version.java]: this requires renaming the vanilla kit to terracotta-5.7... Is this what we really want ? The name collde with the core base kit

Comment (mathieucarbou) [reply common/src/main/java/org/terracotta/angela/common/topology/Version.java]: Also, I disagree that each time we have a version, kit or project we have to update angela.

I would like angela to be able to run on any kit I provide. This is not normal to have hardcoded names and versions in the code IMO.

Comment (saurabhagas) [reply common/src/main/java/org/terracotta/angela/common/topology/Version.java]: The support for 5.7 kit is added in angela. The name of the kit doesn't matter.

>> Also, I disagree that each time we have a version, kit or project we have to update angela.
We don't need to do it every time. Whatever changes were needed in Angela for versioning are done, and won't require further changes for each released version.

>> This is not normal to have hardcoded names and versions in the code IMO.
I'm with you. It's the way angela works. Feel free to rewrite it ;-)

>> We decided to use angela if we were able to use it as a drop in replacement of what we had before
It'd work correctly if properties are correctly set. Let me know if the properties don't work out.

Comment (mathieucarbou) [reply common/src/main/java/org/terracotta/angela/common/topology/Version.java]: angela oss or ee ?

**Comment by saurabhagas** [comment]
Declining in favor of https://github.com/Terracotta-OSS/angela/pull/22

**Pull request closed without merging**